artemis/modules/pdf_censor_scanner.py

```diff
@@ -0,0 +1,45 @@
+from enum import Enum
+from typing import Any, Dict, List
+
+import xray
+from karton.core import Task
+
+from artemis.binds import Service, TaskStatus, TaskType
+from artemis.module_base import ArtemisBase
+from artemis.task_utils import get_target_url
+
+
+class PDFScannerFindings(Enum):
+    CENSORSHIP_WEAKNESS = "censorship_weakness"
+
+
+class PDFCensorScanner(ArtemisBase):
+    """Scans PDFs for censorships using x-ray"""
+
+    identity = "pdf_censor_scanner"
+    filters = [
+        {"type": TaskType.SERVICE.value, "service": Service.HTTP.value},
+    ]
+
+    def process(self, url) -> Dict[int, List[Dict[str, Any]]]:
+        pdf = self.http_get(url).content
+        detected_text = xray.inspect(pdf)
+        return detected_text
+
+    def run(self, current_task: Task) -> None:
+        url = get_target_url(current_task)
+        if url.endswith(".pdf"):
```

Comment:
I'm afraid it would never run, as the URLs are website root URLs and these are never PDF urls. What do you think about

1. crawling for all PDFs on a website (or using open-source sources such as archive.org or Common Crawl - https://github.com/lc/gau)
2. running the check on all PDFs?